### PR TITLE
setup emails

### PR DESCRIPTION
**Pull request by Mattensor**

### CODE REVIEW

File: spec/controllers/users_controller_spec.rb [hunk 1]
```diff
@@ -35,5 +36,21 @@
           expect(assigns(:user)).to be_instance_of(User)
         end
     end
+
+    context "sending emails" do
+      after { ActionMailer::Base.deliveries.clear }
+      it "sends the email to the user with valid inputs" do
+        post :create, user: { email: "eddy@gmail.com", password: "password", full_name: "sir edward"}
+        expect(ActionMailer::Base.deliveries.last.to).to eq(["eddy@gmail.com"])
+      end
```
Comment (tomtomecek) [inline spec/controllers/users_controller_spec.rb]: There is a missing empty line between examples.

File: db/migrate/20150616063719_generate_tokens_for_exisiting_users.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class GenerateTokensForExisitingUsers < ActiveRecord::Migration
+  def change
+    User.all.each do |user|
```
Comment (tomtomecek) [inline db/migrate/20150616063719_generate_tokens_for_exisiting_users.rb]: Schema migrations should not be used as data migrations. These are not reversible to old values. Instead you should use custom rake tasks. Read more here: http://guides.rubyonrails.org/command_line.html#custom-rake-tasks

File: app/views/users/new.html.haml [hunk 3]
```diff
@@ -17,4 +17,4 @@
             = f.text_field :full_name, class: "col-sm-6"
         %fieldset.actions.control-group.col-sm-offset-2
           .controls
-            %input(type="submit" value="Sign Up" class="btn btn-default")
\ No newline at end of file
+            %input(type="submit" value="Sign Up" class="btn btn-default")
```
Comment (tomtomecek) [inline app/views/users/new.html.haml]: using `= f.submit ...` would look more concise. )

File: app/views/forgot_passwords/new.html.haml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    .col-sm-10.col-bg-offset-1
+      = form_tag forgot_passwords_path, class: "sign_in" do
+        %header
+          %h1 Forgot Password?
+          %p We will send you an email with a link that you can use to reset your password.
+        .form-group
+          = label_tag :email, "Email Address"
+          .row
+            .col-sm-4
+              = text_field_tag :email
```
Comment (tomtomecek) [inline app/views/forgot_passwords/new.html.haml]: This should be `email_field_tag`

File: app/views/forgot_passwords/new.html.haml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        .form-group.action
+          %input(type="submit" value="Send Email" class="btn btn-default")
```
Comment (tomtomecek) [inline app/views/forgot_passwords/new.html.haml]: This should be `submit_tag`

File: app/views/password_resets/show.html.haml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+        %header
+          %h1 Reset Your Password
+        %fieldset.form-group
+          = label_tag :password, "New Password"
+          .row
+            .col-sm-4
+              = password_field_tag :password
+              = hidden_field_tag :token, @token
+        %fieldset.form-group.action
+          %input(type="submit" value="Reset Password" class="btn btn-default")
```
Comment (tomtomecek) [inline app/views/password_resets/show.html.haml]: This should be `submit_tag`

File: app/models/user.rb [hunk 7]
```diff
@@ -8,6 +8,8 @@ class User < ActiveRecord::Base
   has_many :following_relationships, class_name: "Relationship", foreign_key: :follower_id
   has_many :leading_relationships, class_name: "Relationship", foreign_key: :leader_id
 
+  before_create :generate_token
```
Comment (tomtomecek) [inline app/models/user.rb]: This should be changed. You should not create a token, when user is created, but when the password reset is requested.

File: app/mailers/app_mailer.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+class AppMailer < ActionMailer::Base
+  def send_welcome_email(user)
+    @user = user
+    mail from: "info@myflix.com", to: user.email, subject: "Welcome to myflix"
```
Comment (tomtomecek) [inline app/mailers/app_mailer.rb]: the `from: "info@myflix.com"` can be extracted as default above, like a callback. `default from: "..."`

File: app/controllers/password_resets_controller.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    user = User.where(token: params[:id]).first
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path 
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
```
Comment (tomtomecek) [inline app/controllers/password_resets_controller.rb]: You should not use `where(arg).first` for fetching single records. It is inefficient. Instead, in Rails 4, it was added `find_by`, which is your primary choice for single record search with ActiveRecord.

File: app/controllers/password_resets_controller.rb [hunk 10]
```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
```
Comment (tomtomecek) [inline app/controllers/password_resets_controller.rb]: You should not use `where(arg).first` for fetching single records. It is inefficient. Instead, in Rails 4, it was added `find_by`, which is your primary choice for single record search with ActiveRecord.

File: app/controllers/forgot_passwords_controller.rb [hunk 11]
```diff
@@ -0,0 +1,15 @@
+class ForgotPasswordsController < ApplicationController
+  def new
+  end
+
+  def create
+    user = User.where(email: params[:email]).first
```
Comment (tomtomecek) [inline app/controllers/forgot_passwords_controller.rb]: You should not use `where(arg).first` for fetching single records. It is inefficient. Instead, in Rails 4, it was added `find_by`, which is your primary choice for single record search with ActiveRecord.

File: db/migrate/20150616063719_generate_tokens_for_exisiting_users.rb (hunk 2 above)
Comment (Mattensor) [inline db/migrate/20150616063719_generate_tokens_for_exisiting_users.rb]: Thanks

**Pull request merged**